authservice/auth_service.go

```diff
@@ -392,14 +545,73 @@ func (r *RemoteAuthDataLoaderImpl) requestDeletedAccounts() ([]string, error) {
 	return deletedAccounts, nil
 }
 
-func (r *RemoteAuthDataLoaderImpl) setupGetDeletedAccountsTimer() {
-	//cancel if active
-	if r.getDeletedAccountsTimer != nil {
-		r.timerDone <- true
-		r.getDeletedAccountsTimer.Stop()
+func (r *RemoteAuthDataLoaderImpl) makeRequest(requestFunc func(*AccessToken) (interface{}, error), token *AccessToken, appOrgPair AppOrgPair, retryString string, updateTokenIfNeeded bool) (interface{}, error) {
```

Comment:
We should expose an interface similar to this to the implementing service to allow them to easily make requests to other services with the appropriate authorization. This interface should allow the implementing service to provide an http.Request, *appID, *orgID, then handle all authorization details (token cache, retries on expired token...). 

If a non-null app/org is passed, we should handle tokens that cover these cases with a null. For example, if both app and org IDs are specified in the params and a service token with null, null app/org IDs is available, we should use that token to make the request.